src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -853,43 +854,60 @@ def generate_flow_tools_json(
     return flow_tools
 
 
-def update_user_agent_from_env_var():
-    """Update user agent from env var to OperationContext"""
-    from promptflow._core.operation_context import OperationContext
+class ClientUserAgentUtil:
+    """SDK/CLI side user agent utilities."""
+
+    @classmethod
+    def _get_context(cls):
+        from promptflow._core.operation_context import OperationContext
+
+        return OperationContext.get_instance()
+
+    @classmethod
+    def get_user_agent(cls):
+        context = cls._get_context()
+        # directly get from context since client side won't need promptflow/xxx.
+        return context.get(USER_AGENT_KEY, "").strip()
+
+    @classmethod
+    def append_user_agent(cls, user_agent: Optional[str]):
+        if not user_agent:
+            return
+        context = cls._get_context()
+        context.append_user_agent(user_agent)
```

Comment:
Should we also modify the fixed string of "user agent" inside the OperaonContext to `USER_AGENT_KEY `?